### PR TITLE
Make sure the match_headers errors are displayed correctly for lower cased version

### LINKED ISSUE #106: Matching the `Authorization` header correctly while another parameter doesn't match causes a very confusing error message
With the following test, which correctly matches the Authorization header, but with an incorrect url in this case, the non-matching request causes a very confusing error message that seems to suggest the headers are wrong (besides the url also being wrong):

```python
from pytest_httpx import HTTPXMock
import httpx


def test_something(httpx_mock: HTTPXMock):
    httpx_mock.add_response(
        method="GET",
        url="http://www.google.com?q=b",
        match_headers={"Authorization": "Bearer: Something"},
    )
    with httpx.Client() as client:
        client.get(
            "http://www.google.com", headers={"Authorization": "Bearer: Something"}
        )
```

Running this with pytest results in the following error:

```
E       httpx.TimeoutException: No response can be found for GET request on http://www.google.com with {} headers amongst:
E       Match GET requests on http://www.google.com?q=b with {'Authorization': 'Bearer: Something'} headers
```

I was expecting it to fail, but it says `with {} header` as if no headers were passed in, while they were.

I found out what the issue is:
In [_explain_that_no_response_was_found](https://github.com/Colin-b/pytest_httpx/blob/develop/pytest_httpx/_httpx_mock.py#L235C116-L235C128) it  includes `if name in expect_headers}`. The issue is that in the request, the `Authorization` header is turned into lowercase. So this ends up on that line as `"authorization" not in {"Authorization": "Bearer: Something"}`, so it won't display the header.

### PR DESCRIPTION
**Pull request by dolfandringa**
closes #106 

Make sure we still display the original header name and value in errors if httpx converted a header name to lower case, and another parameter doesn't match. 

So we display
```python
httpx.TimeoutException: No response can be found for GET request on http://www.google.com with {'MyHeader': 'Something'} headers amongst:
E       Match GET requests on http://www.google.com?q=b with {'MyHeader': 'Something'} headers
```
instead of 
```python
httpx.TimeoutException: No response can be found for GET request on http://www.google.com with {} headers amongst:
E       Match GET requests on http://www.google.com?q=b with {'MyHeader': 'Something'} headers
```
with this test:
```python
from pytest_httpx import HTTPXMock
import httpx


def test_something(httpx_mock: HTTPXMock):
    httpx_mock.add_response(
        method="GET",
        url="http://www.google.com?q=b",
        match_headers={"MyHeader": "Something"},
    )
    with httpx.Client() as client:
        client.get(
            "http://www.google.com", headers={"MyHeader": "Something"}
        )
```

### CODE REVIEW

File: pytest_httpx/_httpx_mock.py [hunk 1]
```diff
@@ -232,7 +232,12 @@ def _explain_that_no_response_was_found(self, request: httpx.Request) -> str:
 
         request_description = f"{request.method} request on {request.url}"
         if expect_headers:
-            request_description += f" with {dict({name: value for name, value in request.headers.items() if name in expect_headers})} headers"
+            present_headers = dict(
+                (name, request.headers.get(name))
+                for name in expect_headers
+                if request.headers.get(name) is not None
+            )
```
Comment (Colin-b) [inline pytest_httpx/_httpx_mock.py:239]:
```suggestion
            present_headers = {
                name: request.headers.get(name)
                for name in expect_headers
                if request.headers.get(name) is not None
            }
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 ### Fixed
+- If httpx lower cases a header name (like authorization) make sure the header is still displayed in the error if another parameter doesn't match
```
Comment (Colin-b) [inline CHANGELOG.md:9]:
```suggestion
- If `httpx` lower cases a header name (like authorization), make sure the header is still displayed in the error if another parameter doesn't match.
```

File: tests/test_httpx_async.py [hunk 3]
```diff
@@ -1116,6 +1116,26 @@ async def test_headers_not_matching(httpx_mock: HTTPXMock) -> None:
     httpx_mock.reset(assert_all_responses_were_requested=False)
 
 
+@pytest.mark.asyncio
+async def test_url_not_matching_upper_case_headers_matching(httpx_mock: HTTPXMock):
```
Comment (Colin-b) [inline tests/test_httpx_async.py:1120]:
```suggestion
async def test_url_not_matching_upper_case_headers_matching(httpx_mock: HTTPXMock) -> None:
```

File: tests/test_httpx_sync.py [hunk 4]
```diff
@@ -139,6 +139,25 @@ def test_response_with_html_string_body(httpx_mock: HTTPXMock) -> None:
         assert response.text == "<body>test content</body>"
 
 
+def test_url_not_matching_upper_case_headers_matching(httpx_mock: HTTPXMock):
```
Comment (Colin-b) [inline tests/test_httpx_sync.py:142]:
```suggestion
def test_url_not_matching_upper_case_headers_matching(httpx_mock: HTTPXMock) -> None:
```

**Review by Colin-b — COMMENTED** [review]
Thanks a lot, I took the liberty to add some cosmetics changes :)

**Comment by Colin-b** [comment]
Also the order of the headers in the displayed message seems to not be consistent (as one test case works with python 3.9 while it fails with python 3.10). It might be good to have a look at this as well :)

**Comment by dolfandringa** [comment]
Thanks for the work @Colin-b

**Comment by Colin-b** [comment]
> Thanks for the work @Colin-b

Back at you :) I will let you know once released, should be done this week

**Comment by Colin-b** [comment]
Thanks again @dolfandringa , release 0.24.0 is now available on pypi

**Pull request merged**